python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -351,24 +352,36 @@ class RandomForestRegressor(Base):
             raise ValueError("Wrong value passed in for max_features"
                              " please read the documentation")
 
-    def _get_model_info(self):
+    def _convert_to_treelite(self, task_category):
         cdef ModelHandle cuml_model_ptr = NULL
-
-        task_category = 1
         cdef RandomForestMetaData[float, float] *rf_forest = \
             <RandomForestMetaData[float, float]*><size_t> self.rf_forest
         build_treelite_forest(& cuml_model_ptr,
                               rf_forest,
                               <int> self.n_cols,
                               <int> task_category,
                               <vector[unsigned char] &> self.model_pbuf_bytes)
-
         mod_ptr = <size_t> cuml_model_ptr
-        fit_mod_ptr = ctypes.c_void_p(mod_ptr).value
+        treelite_handle = ctypes.c_void_p(mod_ptr).value
+        return treelite_handle
+
+    def _get_model_info(self):
+        task_category = 1
+        fit_mod_ptr = self._convert_to_treelite(task_category)
         cdef uintptr_t model_ptr = <uintptr_t> fit_mod_ptr
         model_protobuf_bytes = save_model(<ModelHandle> model_ptr)
         return model_protobuf_bytes
 
+    def convert_to_treelite_model(self):
```

Comment:
There seems to be a lot of duplicate code between `RandomForestClassifier` and `RandomForestRegressor`. Consider moving the common code into a base class. This is definitely for a separate pull request, however.

Reply:
Agreed on both counts - separate PR but still needed.

Reply:
Will create a PR to reduce the duplicate code